File: power-of-three/power-of-three.py
class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 1:
            return True
        if n %3 != 0 or n == 0:
            return False
        return self.isPowerOfThree(n/3)

File: power-of-three/power-of-three.java
import java.lang.*;

class Solution {
    public boolean isPowerOfThree(int n) {
        
        if(n == 1){
            return true;
        }
        if (n <= 0 || n%3!=0){
            return false;
        }
            
        double a = Math.log10(n)/Math.log10(3);
        int b = (int)a;
        return a==b;
    }
}
